rcl/test/rcl/test_namespace.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <chrono>
+#include <thread>
+
+#include "rcl/client.h"
+
+#include "rcl/rcl.h"
+#include "rcl/graph.h"
+
+#include "example_interfaces/srv/add_two_ints.h"
+#include "rosidl_generator_c/string_functions.h"
+
+#include "../memory_tools/memory_tools.hpp"
+#include "../scope_exit.hpp"
+#include "rcl/error_handling.h"
+
+using namespace std::chrono_literals;
+
+class TestNamespaceFixture : public ::testing::Test
+{
+public:
+  rcl_node_t * node_ptr;
+  void SetUp()
+  {
+    stop_memory_checking();
+    rcl_ret_t ret;
+    ret = rcl_init(0, nullptr, rcl_get_default_allocator());
+    ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe();
+    this->node_ptr = new rcl_node_t;
+    *this->node_ptr = rcl_get_zero_initialized_node();
+    const char * name = "node_name";
```

Comment:
Since you changed the name in `test_graph.cpp`, maybe this should be something like `test_namespace_node`?